src/main/scala/exercise2/lib/Synthesizer.scala

```diff
@@ -0,0 +1,64 @@
+package exercise2.lib
+
+import exercise2.{QueueLike, Queue}
+
+import scala.annotation.tailrec
+import scala.util.Random
+
+object Synthesizer {
+  val ENERGY_DECAY_FACTOR = 0.996
+
+  /**
+   * Karplus-Strong Part 1
+   *
+   * Create a function def whiteNoise(frequency:Int=440, volume:Double = 1.0): Queue[Double]
+   * that returns a queue containing a total of frequency elements of random values between .5 and -.5 multiplied by volume.
+   * Frequency must be greater than zero and volume is between 0 and 1.
+   * */
+  def whiteNoise(frequency: Int = 440, volume: Double = 1.0): QueueLike[Double] = {
+    if (frequency <= 0 || volume < 0 || volume > 1) {
+      throw new IllegalArgumentException
+    }
+
+    @tailrec
+    def fillQueue(queue: QueueLike[Double], count: Int): QueueLike[Double] = {
+      if (count == 0) return queue
+
+      val newQueue = queue.enqueue(Random.between(-0.5, 0.5) * volume)
+      fillQueue(newQueue, count - 1)
+    }
+
+    fillQueue(new Queue[Double](), frequency)
+  }
+
+  /**
+   * Karplus-Strong Part 2
+   *
+   * Remove the front double in the queue and average it with the next double
+   * (hint: use dequeue() and peek()) multiplied by an energy decay factor of 0.996 (we’ll call this entire quantity newDouble).
+   * Then, add newDouble to the queue.
+   *
+   * This function should be pure - it takes a queue and returns a new queue
+   * */
+  def update(queue: QueueLike[Double]): QueueLike[Double] = {
+    val newQueue = queue.dequeue().get
```

Comment:
`.get` on `Option` should be avoided. would be nicer if `update` itself would return `Option[QueueLike[Double]]` and then end the end, when really needed, call `.get` on that `Option`. (there are ways to avoid called that get as well using IO, which we will check out in this week's lecture)